domain/application/service/unit.go

```diff
@@ -564,23 +573,36 @@ func (s *Service) GetUnitWorkloadVersion(ctx context.Context, unitName coreunit.
 // - [uniterrors.UnitNotFound] if the unit does not exist
 // - [network.NoAddressError] if the unit has no public address associated
 func (s *Service) GetUnitPublicAddress(ctx context.Context, unitName coreunit.Name) (network.SpaceAddress, error) {
-	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
+	publicAddresses, err := s.GetUnitPublicAddresses(ctx, unitName)
 	if err != nil {
 		return network.SpaceAddress{}, errors.Capture(err)
 	}
+	return publicAddresses[0], nil
+}
+
+// GetUnitPublicAddresses returns all public addresses for the specified unit.
+//
+// The following errors may be returned:
+// - [uniterrors.UnitNotFound] if the unit does not exist
+// - [network.NoAddressError] if the unit has no public address associated
+func (s *Service) GetUnitPublicAddresses(ctx context.Context, unitName coreunit.Name) (network.SpaceAddresses, error) {
+	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
 	addrs, err := s.st.GetUnitAddresses(ctx, unitUUID)
```

Comment:
Should this be swapped over to `GetUnitPublicAddresses`?

Reply:
Sorry what do you mean? This line is part of `GetUnitPublicAddresses()` already.

Reply:
Oops thing I got things confused